File: backend/models/resume.py
from pydantic import BaseModel
from typing import Optional

class Form(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    location: Optional[str] = None
    linkedin: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    website: Optional[str] = None

class ProfileRequest(BaseModel):
    github_user: Optional[str] = None
    leetcode_user: Optional[str] = None
    bootdev_user: Optional[str] = None
    summarize: Optional[bool] = False
    personal: Optional[Form] = None

File: frontend/src/types/resume.ts
export interface PersonalInfo {
    first_name?: string;
    last_name?: string;
    email?: string;
    phone?: string;
    location?: string;
    linkedin?: string;
    github?: string;
    twitter?: string;
    website?: string;
}

export interface ResumeGenReq {
    github_user?: string;
    leetcode_user?: string;
    bootdev_user?: string;
    summarize?: boolean;
    personal?: PersonalInfo;
}

export interface ResumeGenRsp {
    status: string;
    message: string;
    docx_filename: string;
    resume_id: string;
}
